src/haz3lcore/statics/MakeTerm.re

```diff
@@ -259,7 +260,21 @@ and exp_term: unsorted => (UExp.term, list(Id.t)) = {
     | ([(_id, t)], []) =>
       ret(
         switch (t) {
-        | (["(", ")"], [Exp(arg)]) => Ap(l, arg)
+        | (["(", ")"], [Exp(arg)]) =>
+          let is_deferral = (e: Term.UExp.t) =>
+            switch (e.term) {
+            | Deferral => true
+            | _ => false
+            };
+          switch (arg.term) {
+          | Tuple(es) =>
```

Comment:
this case can be replaced by:

| Tuple(es) when List.exists(is_deferral, es) => DeferredAp(l, es)